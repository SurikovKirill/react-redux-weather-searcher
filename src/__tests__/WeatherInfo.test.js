import WeatherInfo from '../components/WeatherInfo';
import renderer from 'react-test-renderer';
import React from 'react';

describe('WeatherInfo component', () => {

    test('matches the snapshot when weather != null', () => {
        const city = {
            name: 'Moscow',
            img: 'i',
            temperature: 't',
            wind: 'w',
            cloudiness: 'c',
            pressure: 'p',
            humidity: 'h',
            location: 'l'
        };
        const tree = renderer.create(
           <WeatherInfo weather={city}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot when weather == null', () => {

        const tree = renderer.create(
            <WeatherInfo weather={null}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});