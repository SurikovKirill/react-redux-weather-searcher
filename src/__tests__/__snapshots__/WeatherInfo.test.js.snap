// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo component matches the snapshot when weather != null 1`] = `
<div
  className="WeatherInfo"
>
  <div
    className="Info"
  >
    <h3>
      Ветер
    </h3>
    <h4>
      w
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Облачность
    </h3>
    <h4>
      c
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Давление
    </h3>
    <h4>
      p
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Влажность
    </h3>
    <h4>
      h
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Координаты
    </h3>
    <h4>
      l
    </h4>
  </div>
</div>
`;

exports[`WeatherInfo component matches the snapshot when weather == null 1`] = `null`;
